provider/azure/credentials.go

```diff
@@ -65,12 +77,50 @@ func (environProviderCredentials) CredentialSchemas() map[cloud.AuthType]cloud.C
 				},
 			},
 		},
+
+		// azureCLIAuthType creates the credential using access
+		// tokens retrieved from the Azure CLI application. This
+		// is only supported on the client side; it will be used
+		// to generate a service principal, and transformed into
+		// clientCredentialsAuthType.
+		azureCLIAuthType: {},
 	}
 }
 
-// DetectCredentials is part of the environs.ProviderCredentials interface.
-func (environProviderCredentials) DetectCredentials() (*cloud.CloudCredential, error) {
-	return nil, errors.NotFoundf("credentials")
+// DetectCredentials is part of the environs.ProviderCredentials
+// interface. It attempts to detect subscription IDs from accounts
+// configured in the Azure CLI.
+func (c environProviderCredentials) DetectCredentials() (*cloud.CloudCredential, error) {
+	// Attempt to get accounts from az.
+	accounts, err := c.azListAccounts()
+	if err != nil {
+		logger.Debugf("error getting accounts from az: %s", err)
```

Comment:
not unless we're going to be more meticulous about checking az compatibility, particularly checking for "get-access-token" support